### DISCUSSION TITLE
Repository Group By error

### DISCUSSION THREAD

**Discussion by DavideCarvalho** [mikro-orm/mikro-orm#4459] [Q&A]
**Describe the bug**
If I use "groupBy" with a repository, it breaks

![image](https://github.com/mikro-orm/mikro-orm/assets/15630886/3e5f67a6-6937-4a3e-93fd-807f92aa1c5e)

![image](https://github.com/mikro-orm/mikro-orm/assets/15630886/a52b3b88-3ad8-407d-9b4f-4553c7c6e0f0)

**Stack trace**
```
"select \"d0\".* from \"device_transmission_revision\" as \"d0\" where \"d0\".\"id\" = 'd87d8cd1-0e70-4e6d-a299-c0ab9fbe98e9' group by \"d0\".\"owner_id\" limit 1 - column \"d0.id\" must appear in the GROUP BY clause or be used in an aggregate function"
```

**To Reproduce**
1. Have a repository
2. Do a find
3. Add "groupBy" into the options arg of the find

**Expected behavior**
Group result by the field

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 20.2.0 | 
| typescript | 4.9.3 | 
| mikro-orm | 5.7.12 | 
| mikro-orm postgres | 5.7.12 |

**Comment by B4nan** [comment]
The error says it all, you can't use group by clause when you select all the fields, that's how SQL grouping works. The option there is to allow you to control the group by clause, it does not affect anything else, it does return "grouped entity results" anyhow, it's really just about adding the SQL fragment - it's up to you to use it properly.

Not sure what you are trying to do, why would you want to add `group by` to a `findOne` query? It will always return one result.

**Reply by DavideCarvalho** [reply]
Oh sorry, the final code doesn't have a findOne lol

I was trying to do a find and wanted to group the results by owner

I thought that MikroORM would change the query by itself to allow the groupBy, but it looks like it doesn't

So, is there a workaround for that? If I want to do a find with a groupBy, what should I do?

**Reply by B4nan** [reply]
The real question is what are you expecting as the result, as em.find will always give you flat array of entities, there is no grouping. 

Maybe we should have something like  em.findAndGroupBy() which would return a dictionary instead of array.
